fix(semantic): incorrect `scope_id` for catch parameter symbols

Dunqing opened the pull request:
When we move all the bindings in the CatchClause scope to its child scope (BlockStatement), we also need to replace the scope_id in the symbol with the scope_id of the new scope.

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4659?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4659** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4659?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#4658** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4658?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (commented):
Merge conflicts. But this change looks good.

Although, I have to ask... why did we add a scope for catch params since we now never attach any bindings to it?

Boshen commented:
Merge conflict.

Dunqing commented:
> Although, I have to ask... why did we add a scope for catch params since we now never attach any bindings to it?

See https://github.com/oxc-project/oxc/issues/4186

Dunqing committed:
fix(semantic): incorrect `scope_id` for catch parameter symbols (#4659)

When we move all the bindings in the CatchClause scope to its child scope (BlockStatement), we also need to replace the scope_id in the symbol with the scope_id of the new scope.
